mdtools.py

```diff
@@ -26,17 +34,36 @@ def grompp(
     if grompp_options is None:
         grompp_options = []
     output_files = {"-o": tpr_file, "-po": mdp_output_file}
-    prep = gmx.commandline_operation(
-        executable="gmx",
-        arguments=["grompp"] + grompp_options,
-        input_files=input_files,
-        output_files=output_files,
-    )
-    prep.run()
-    output = str(prep.output.erroroutput.result()).strip()
+    if use_api:
+        prep = gmx.commandline_operation(
+            executable="gmx",
+            arguments=["grompp"] + grompp_options,
+            input_files=input_files,
+            output_files=output_files,
+        )
+        prep.run()
+        output = str(prep.output.erroroutput.result()).strip()
+    else:
+        infiles = ' '.join([k + ' ' + v for k, v in input_files.items()])
+        outfiles = ' '.join([k + ' ' + v for k, v in output_files.items()])
+        gmx = "mpiexec -n 1 gmx_mpi" if gmx_executable == "gmx_mpi" else "gmx"
+        parse_options =  ' '.join(grompp_options)
+        result = run(f"{gmx} grompp {parse_options} {infiles} {outfiles}", stdout=PIPE, stderr=PIPE, shell=True)
+        output = result.stderr
+
     if output:
         logger.info("grompp output:\n%s", output)
-    return prep
+
+
+def grompp_all(task_list: List[dict]):
+    from multiprocessing import Pool
+    proc_per_core = int(os.environ["SLURM_CPUS_ON_NODE"]) if "SLURM_CPUS_ON_NODE" in os.environ.keys() else 1
+    with Pool(proc_per_core) as p:
+        p.map(_grompp, task_list)
+
+
+def _grompp(args: dict):
```

Comment:
args.update({'gmx_executable': 'gmx_mpi'}) in my system returns None and does the update to args. Maybe this has changed from python version (I'm using 3.8.5).
I suggest:
```python
def _grompp(args: dict):
    args.update({"gmx_executable": "gmx_mpi"})
    return grompp(**args)
```